Critical: Missing svelte-dnd-action module causes 500 error on startup

## Description

After pulling the latest main branch, the application fails to start with a 500 error due to a missing module.

## Error

```
[500] GET /
Error: Cannot find module 'svelte-dnd-action' imported from '/Users/evanmiller/git/director-assist/src/lib/components/layout/Sidebar.svelte'
```

## Root Cause

The `svelte-dnd-action` dependency is listed in `package.json` but may not be installed in users' `node_modules` after pulling new changes. This is standard npm behavior but can be confusing.

## Fix

Add documentation to README reminding users to run `npm install` after pulling changes, especially when dependency updates are included.

## Acceptance Criteria

- [ ] README includes a note in the Development section about reinstalling dependencies after pulling
- [ ] CHANGELOG mentions any new dependencies added in releases

## Workaround

```bash
npm install
```

## Priority

Low - Documentation improvement

## Affected Version

v0.5.0+

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -10,6 +10,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [0.5.0] - 2026-01-19
 
 ### Added
+
+**New Dependencies**: `svelte-dnd-action` - Run `npm install` after upgrading to this version.
 - Custom entity types with full field customization (Issue #25, PR #169)
   - Create custom entity types with 14 field types: text, number, date, boolean, select, multiselect, currency, percentage, rating, richtext, file, image, entity-ref, entity-refs
   - Computed fields with dynamic formula evaluation and automatic type inference

--- a/README.md
+++ b/README.md
@@ -35,6 +35,8 @@ npm run dev
 
 Open your browser to `http://localhost:5173`
 
+**After pulling updates**: Always run `npm install` after pulling new changes to ensure any new dependencies are installed.
+
 ### Building for Production
 
 ```bash

